### PR TITLE
[tabulator-tables] support array args for groupBy option

### PR DESCRIPTION
**Pull request by tplk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://tabulator.info/docs/4.8/group#multi (Multi Level Grouping)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/tabulator-tables/index.d.ts [hunk 1]
```diff
@@ -1271,6 +1271,7 @@ You can pass an optional additional property with sorter, sorterParams that shou
     }
 
     type GroupValuesArg = any[][];
+    type GroupByArg = string | string[] | ((data: any) => any) | Array<((data: any) => any)>;
```
Comment (Jojoshua) [inline types/tabulator-tables/index.d.ts:1274]: Can you provide an example and test/docs of where this is used? I didn't see where you can pass in an array of functions

Comment (Jojoshua) [inline types/tabulator-tables/index.d.ts:1274]: This makes sense

**Review by Jojoshua — CHANGES_REQUESTED** [review]

File: types/tabulator-tables/index.d.ts (hunk 1 above)
Comment (tplk) [reply types/tabulator-tables/index.d.ts:1274]: Hello! It's not stated explicitly in the docs but you can validate it in the source code:

- https://github.com/olifolkerd/tabulator/blob/master/src/js/modules/group_rows.js#L819-L821 - here a single value is coerced into an array.

- https://github.com/olifolkerd/tabulator/blob/master/src/js/modules/group_rows.js#L826 - a couple lines after that there's a check if a function was passed.

Comment (tplk) [reply types/tabulator-tables/index.d.ts:1274]: It's actually a good point that the docs don't mention this, I'll create a request for the docs change too.

Comment (Jojoshua) [reply types/tabulator-tables/index.d.ts:1274]: Ok thanks, I wonder if that is legacy code or meant for the public API. I'm definitely good with the `string[] `  change

**Pull request closed without merging**